spec/API_specification/signatures/creation_functions.py

```diff
@@ -0,0 +1,371 @@
+        -   if one or more values are ``float``s, the output data type must be the default floating-point data type.
+
+        Default: ``None``.
+
+        **Note**
+
+        If ``dtype`` is not ``None``, then array conversions should obey :ref:`type-promotion` rules. Conversions not specified according to :ref:`type-promotion` rules may or may not be permitted by a conforming array library.
+        To perform an explicit cast, use :ref:`function-astype`.
+    device: Optional[device]
+        device on which to place the created array. If ``device`` is ``None`` and ``x`` is an array, the output array device must be inferred from ``x``. Default: ``None``.
+    copy: Optional[bool]
+        boolean indicating whether or not to copy the input. If ``True``, the function must always copy. If ``False``, the function must never copy for input which supports the buffer protocol and must raise a ``ValueError`` in case a copy would be necessary. If ``None``, the function must reuse existing memory buffer if possible and copy otherwise. Default: ``None``.
+
+    Returns
+    -------
+    out: array
+        an array containing the data from ``obj``.
+    """
+
+def empty(shape: Union[int, Tuple[int, ...]], *, dtype: Optional[dtype] = None, device: Optional[device] = None) -> array:
+    """
+    Returns an uninitialized array having a specified `shape`.
+
+    Parameters
+    ----------
+    shape: Union[int, Tuple[int, ...]]
+        output array shape.
+    dtype: Optional[dtype]
+        output array data type. If ``dtype`` is ``None``, the output array data type must be the default floating-point data type. Default: ``None``.
+    device: Optional[device]
+        device on which to place the created array. Default: ``None``.
+
+    Returns
+    -------
+    out: array
+        an array containing uninitialized data.
+    """
+
+def empty_like(x: array, /, *, dtype: Optional[dtype] = None, device: Optional[device] = None) -> array:
+    """
+    Returns an uninitialized array with the same ``shape`` as an input array ``x``.
+
+    Parameters
+    ----------
+    x: array
+        input array from which to derive the output array shape.
+    dtype: Optional[dtype]
+        output array data type. If ``dtype`` is ``None``, the output array data type must be inferred from ``x``. Default: ``None``.
+    device: Optional[device]
+        device on which to place the created array. If ``device`` is ``None``, the output array device must be inferred from ``x``. Default: ``None``.
+
+    Returns
+    -------
+    out: array
+        an array having the same shape as ``x`` and containing uninitialized data.
+    """
+
+def eye(n_rows: int, n_cols: Optional[int] = None, /, *, k: int = 0, dtype: Optional[dtype] = None, device: Optional[device] = None) -> array:
+    """
+    Returns a two-dimensional array with ones on the ``k``th diagonal and zeros elsewhere.
```

Comment:
Sphinx doesn't like this. I think you can fix it with ``` ``k``\th ``` (just make sure to make the docstring raw). See https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#character-level-inline-markup-1

This is also a problem in some other places. See the warnings when building.